src/mcts/search.cc

```diff
@@ -2022,7 +2032,7 @@ SearchWorker::NodeToProcess SearchWorker::PickNodeToExtend(
     EnsureNodeTwoFoldCorrectForDepth(node, depth - 1);
 
     // If terminal, we reached the end of this playout.
-    if (node->IsTerminal()) {
+    if (!is_root_node && node->IsTerminal()) {
```

Comment:
This is needed because root can be a terminal now, and we don't want to make it non terminal. Might need to doublecheck the behaviour of navigating through proven mate lines.